app/models/manageiq/providers/azure/cloud_manager/event_catcher/stream.rb

```diff
@@ -2,13 +2,10 @@ class ManageIQ::Providers::Azure::CloudManager::EventCatcher::Stream
   #
   # Creates an event monitor
   #
-  def initialize(client_id, client_key, azure_region, tenant_id, subscription_id)
-    @client_id         = client_id
-    @client_key        = client_key
-    @tenant_id         = tenant_id
-    @azure_region      = azure_region
-    @subscription_id   = subscription_id
+  def initialize(ems)
+    @ems = ems
```

Comment:
This change would force the EventCatcher::Stream to run in a Rails context (because it needs an EMS object).  It was written the way it was to allow it to be used with just raw parameters, and thus could be used outside of Rails.

I'm not against this change, I just want to make sure you were aware of that distinction, and if that was ok with you.